httpapi/v2/converters.go

```diff
@@ -21,6 +21,20 @@ func toDocumentsPayload(req DocumentRequest, docID []byte) (payload documents.Up
 	return documents.UpdatePayload{CreatePayload: cp, DocumentID: docID}, nil
 }
 
+func toCloneDocumentsPayload(req DocumentRequest, docID []byte) (payload documents.UpdatePayload, err error) {
+	cp, err := coreapi.ToDocumentsCreatePayload(req)
+	if err != nil {
+		return payload, err
+	}
+	for _, a := range cp.Attributes {
+		if a.KeyLabel == "template" {
```

Comment:
Following on Ved's comment above, adding the template as an attribute will make the payload susceptible of a name collision. Since we have an specific DeriveFromClonePayload I would rather pass the separate template id as an argument.